### ISSUE TITLE
Wrong type string underlying value printer

### ISSUE THREAD

**Issue by SylvainCorlay** [root-project/cling#228]
Minimal reproducer:

```cpp
namespace ns {
    class param
    {
    };

    template <class TB>
    class temp
    {
    };
}

auto foo = ns::temp<ns::param>();
foo
```

yields

```
input_line_7:2:31: error: unknown type name 'param'; did you mean 'ns::param'?
 cling::printValue(*(ns::temp<param>**)0x7ffee8500df0);
                              ^~~~~
                              ns::param
input_line_3:2:11: note: 'ns::param' declared here
    class param
          ^
ERROR in cling's callPrintValue(): cannot pass value!
(ns::temp<param> &) ERROR in cling's callPrintValue(): missing value string.
```

It seems that `getTypeString` misses some namspace specifiers.

**Comment by SylvainCorlay** [comment]
Note: #225 fixes this minimal reproducer, however, the following is not fixed by the PR:

**Comment by Axel-Naumann** [comment]
```
[cling]$ #include <vector>
[cling]$ namespace b {
[cling]$ ?   
[cling]$ ?       template <class T>
[cling]$ ?       class Foo
[cling]$ ?       {
[cling]$ ?          };
[cling]$ ?   
[cling]$ ?       template <template <class> class T>
[cling]$ ?       class Bar
[cling]$ ?       {
[cling]$ ?          };
[cling]$ ?       
[cling]$ ?       using Baz = Bar<Foo>;
[cling]$ ?   }
[cling]$ 
[cling]$ auto h = std::vector<b::Baz>();
[cling]$ h
(std::vector<b::Bar<b::Foo>, std::allocator<b::Bar<b::Foo> > > &) {}
```

This was fixed by upgrading the type printing ea4efd2. Thanks for your report!
